### PR TITLE
Adds retry logic when fetching conference numbers and pin

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: react/features/base/util/httpUtils.js [hunk 1]
```diff
@@ -25,4 +35,17 @@ export function doGetJSON(url) {
...
+            .catch(response => {
+                if (response.status >= 400 && response.status < 500) {
+                    return Promise.reject(response);
+                }
+
+                return timeoutPromise(fetchPromise, RETRY_TIMEOUT);
+            });
+    }
+
+    return fetchPromise;
```
Comment (virtuacoplenny) [inline react/features/base/util/httpUtils.js]: We have this in jitsi-meet-spot as well. You want to shove something into js-utils so both repos can share it?

File: react/features/base/util/httpUtils.js [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
  *
  * @param {string} url - The URL to perform a GET against.
+ * @param {?boolean} retry - Whether the request will be retried after short timeout.
  * @returns {Promise<Object>} The response body, in JSON format, will be
  * through the Promise.
  */
-export function doGetJSON(url) {
-    return fetch(url)
+export function doGetJSON(url, retry) {
+    const fetchPromise = fetch(url)
```
Comment (virtuacoplenny) [inline react/features/base/util/httpUtils.js]: What this method does below is confusing t ome. Why is it in the jsonify that it's doing Promise.reject on successful jsonify?

Comment (damencho) [reply react/features/base/util/httpUtils.js]: Cause the response was not successful, did not return 200 OK.

File: react/features/base/util/httpUtils.js (hunk 1 above)
Comment (damencho) [reply react/features/base/util/httpUtils.js]: Ok, I will check that.

File: react/features/base/util/httpUtils.js (hunk 2 above)
Comment (virtuacoplenny) [reply react/features/base/util/httpUtils.js]: Okay, I think that's confusing and could easily be cleaned up by moving jsonify its own chained call, that way there is always one return, but you didn't change it so we'll let it be.

Comment (virtuacoplenny) [reply react/features/base/util/httpUtils.js]: Damn, I wrote this mess. What the hell is wrong with my dumbass?

**Pull request merged**